src/sonic-config-engine/portconfig.py

```diff
@@ -1,33 +1,138 @@
 #!/usr/bin/env python
-import os
-import sys
+try:
+    import os
+    import sys
+    import json
+    import ast
+    import re
+    from collections import OrderedDict
+    from swsssdk import ConfigDBConnector
+except ImportError as e:
+    raise ImportError("%s - required module not found" % str(e))
 
+# Global Variable
+PLATFORM_ROOT_PATH = '/usr/share/sonic/device'
+PLATFORM_ROOT_PATH_DOCKER = '/usr/share/sonic/platform'
+SONIC_ROOT_PATH = '/usr/share/sonic'
+HWSKU_ROOT_PATH = '/usr/share/sonic/hwsku'
+
+PLATFORM_JSON = 'platform.json'
+PORT_CONFIG_INI = 'port_config.ini'
+HWSKU_JSON = 'hwsku.json'
+
+PORT_STR = "Ethernet"
+BRKOUT_MODE = "default_brkout_mode"
+CUR_BRKOUT_MODE = "brkout_mode"
+INTF_KEY = "interfaces"
+
+BRKOUT_PATTERN = r'(\d{1,3})x(\d{1,3}G)(\[\d{1,3}G\])?(\((\d{1,3})\))?'
+
+
+#
+# Helper Functions
+#
+def readJson(filename):
+    # Read 'platform.json' or 'hwsku.json' file
+    try:
+        with open(filename) as fp:
+            try:
+                data = json.load(fp)
+            except json.JSONDecodeError:
+                print("Json file does not exist")
+        data_dict = ast.literal_eval(json.dumps(data))
+        return data_dict
+    except Exception as e:
+        print("error occurred while parsing json:", sys.exc_info()[1])
+        return None
+
+def db_connect_configdb():
+    """
+    Connect to configdb
+    """
+    config_db = ConfigDBConnector()
+    if config_db is None:
+        return None
+    try:
+        config_db.db_connect('CONFIG_DB')
```

Comment:
If you would like, we could make the async nature of this call, explicit. Even a comment, as Xu explained, could be helpful for future code readers, to help understand, why async mode is selected.

config_db.connect(wait_for_init=False)

Xu's comments that could be re-phrased
"_This could be blocking during the config load_minigraph phase, as the CONFIG_DB_INITIALIZED is not yet set in the configDB. We can ignore the check by using config_db.db_connect('CONFIG_DB') instead_"

Reply:
Modified as requested!